bgpd/bgp_vty.c

```diff
@@ -1543,17 +1536,19 @@ DEFUN (no_router_bgp,
 			return CMD_WARNING_CONFIG_FAILED;
 		}
 	} else {
-		as = strtoul(argv[idx_asn]->arg, NULL, 10);
+		if (!str2asn(instasn, &as))
+			return CMD_WARNING_CONFIG_FAILED;
+
+		local_name = (char *)name;
```

Comment:
casting to the same type is not necessary: `local_name = name` is enough.

Reply:
reverted to old defun format. actually, two different things in the same commit is not good. so, this will be reverted too.